flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/plan/utils/RexNodeNestedFieldTest.scala

```diff
@@ -0,0 +1,234 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.plan.utils
+
+import org.apache.flink.table.planner.calcite.FlinkRexBuilder
+
+import java.util
+import java.util.{List => JList}
+
+import org.junit.Assert.{assertThat, assertTrue}
+import org.junit.Test
+
+import org.hamcrest.{BaseMatcher, Description}
+
+import scala.collection.JavaConversions._
+import scala.collection.JavaConverters._
+
+/**
+ *  Test for RexNodeNestedField.
+ */
+class RexNodeNestedFieldTest extends RexNodeTestBase{
+  private class UnorderedArrayMatcher(expected: util.HashSet[JList[Int]])
+    extends BaseMatcher[Array[Array[Int]]] {
+    override def matches(item: Any): Boolean = {
+      for (path <- item.asInstanceOf[Array[Array[Int]]]) {
+        if (!expected.contains(path.toSeq.asJava)) {
+          return false
+        }
+      }
+      true
+    }
+
+    override def describeTo(description: Description): Unit = {
+      description.appendValueList("", ",", "", expected)
+    }
+  }
+
+  @Test
+  def testExtractRefInputFields(): Unit = {
+    val (exprs, rowType) = buildExprs()
+    val nestedFields = RexNodeNestedFields.build(exprs, rowType)
+    val actual = RexNodeNestedFields.labelAndConvert(nestedFields)
+    val expected = new util.HashSet[JList[Int]]
+    expected.add(util.Arrays.asList(2))
+    expected.add(util.Arrays.asList(3))
+    expected.add(util.Arrays.asList(1))
+
+    assertThat(actual, new UnorderedArrayMatcher(expected))
+  }
+
+  @Test
+  def testExtractRefNestedInputFields(): Unit = {
+    val (rexProgram, rowType) = buildExprsWithNesting()
+
+    val nestedFields = RexNodeNestedFields.build(rexProgram, rowType)
+    val actual = RexNodeNestedFields.labelAndConvert(nestedFields)
+    val expected = new util.HashSet[JList[Int]]
+    expected.add(Array(1, 1).toSeq.asJava)
+    expected.add(Array(0).toSeq.asJava)
```

Comment:
the expected is  
1,1  
0, 
0.
we can use list or array to store the expected result, and before comparing, we can sort the expected collection and actual collection.